### DISCUSSION TITLE
How to use augumentation parameters in hyp.scratch-low.yaml for yolo11?

### DISCUSSION THREAD

**Discussion by xhuan8** [ultralytics/ultralytics#17461] [General]
In yolov5 there is a argument "--hyp" default to 'data/hyps/hyp.scratch-low.yaml',
Is it possbile to use the same config file in yolo11?

```
from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO("yolo11s.pt")
    model.train(data=r"E:\Dataset\TSIEdge\dataset.yaml", epochs=3, imgsz=2736, batch=1, mosaic=0.0, workers=0)
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xhuan8, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. These resources may help answer some of your questions about using configuration files in YOLO11.

If your question involves custom training configurations, please provide as much detail as possible, including your current YAML files and any relevant training logs. This will help us to assist you more effectively. 

If this is a 🐛 Bug Report, sharing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) would be very helpful.

For real-time chat and community interactions, you might want to visit our [Discord](https://ultralytics.com/discord) 🎧. And for more in-depth discussions, head over to [Discourse](https://community.ultralytics.com). You can also join conversations on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure you are using the latest `ultralytics` package with all dependencies updated. Verification that your issue is not already resolved in the latest version is important:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in several up-to-date verified environments. Try it in a **notebook** with free GPU by clicking below:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

An Ultralytics engineer will also assist you soon to provide more personalized help. Thanks again for reaching out!

**Comment by glenn-jocher** [comment]
@xhuan8 yes, you can use a custom hyperparameters file in YOLO11 by specifying it with the `hyp` argument in the `train` method, similar to YOLOv5.

**Reply by xhuan8** [reply]
I got ''hyp' is not a valid YOLO argument', anything wrong?
```
model.train(data=r"E:\Dataset\TSIEdge\dataset.yaml", epochs=3, imgsz=2736, batch=1, mosaic=0.0, workers=0, hyp="hyp.scratch-low.yaml")
```
```
Traceback (most recent call last):
  File "E:\Code\yolov8\main.py", line 5, in <module>
    model.train(data=r"E:\Dataset\TSIEdge\dataset.yaml", epochs=3, imgsz=2736, batch=1, mosaic=0.0, workers=0, hyp="hyp.scratch-low.yaml")
  File "D:\anaconda3\envs\yolo\Lib\site-packages\ultralytics\engine\model.py", line 796, in train
    self.trainer = (trainer or self._smart_load("trainer"))(overrides=args, _callbacks=self.callbacks)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\anaconda3\envs\yolo\Lib\site-packages\ultralytics\engine\trainer.py", line 101, in __init__
    self.args = get_cfg(cfg, overrides)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\anaconda3\envs\yolo\Lib\site-packages\ultralytics\cfg\__init__.py", line 296, in get_cfg
    check_dict_alignment(cfg, overrides)
  File "D:\anaconda3\envs\yolo\Lib\site-packages\ultralytics\cfg\__init__.py", line 481, in check_dict_alignment
    raise SyntaxError(string + CLI_HELP_MSG) from e
SyntaxError: 'hyp' is not a valid YOLO argument. 
```

**Reply by glenn-jocher** [reply]
@xhuan8 the `hyp` argument is not directly supported in YOLO11. Instead, you can modify hyperparameters directly in the model's configuration file or pass them as individual arguments in the `train` method.

**Reply by xhuan8** [reply]
Thanks, pass argument in train method can work, but put in configuration file didn't take effect.

```
model = YOLO("yolo11s.yaml").load("yolo11s.pt")
```
 "anaconda3\envs\yolo\Lib\site-packages\ultralytics\cfg\models\11\yolo11.yaml",
```
# YOLO11n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 2, C3k2, [512, False]] # 13

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 2, C3k2, [256, False]] # 16 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 13], 1, Concat, [1]] # cat head P4
  - [-1, 2, C3k2, [512, False]] # 19 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 10], 1, Concat, [1]] # cat head P5
  - [-1, 2, C3k2, [1024, True]] # 22 (P5/32-large)

  - [[16, 19, 22], 1, Detect, [nc]] # Detect(P3, P4, P5)
  
lr0: 0.01  # initial learning rate (SGD=1E-2, Adam=1E-3)
lrf: 0.01  # final OneCycleLR learning rate (lr0 * lrf)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
warmup_epochs: 3.0  # warmup epochs (fractions ok)
warmup_momentum: 0.8  # warmup initial momentum
warmup_bias_lr: 0.1  # warmup initial bias lr
```

**Reply by glenn-jocher** [reply]
@xhuan8 to ensure changes in the configuration file take effect, verify that the file path is correct and that the model is loaded with the updated configuration before training.
